presto-spi/src/main/java/com/facebook/presto/spi/security/SslContextProvider.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.airlift.security.pem.PemReader;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static java.security.KeyStore.getDefaultType;
+import static java.security.KeyStore.getInstance;
+import static java.util.Collections.list;
+import static javax.net.ssl.TrustManagerFactory.getDefaultAlgorithm;
+
+/**
+ * Common SSL context provider for Presto connectors.
+ * Handles both PEM and Java KeyStore formats for keystores and truststores.
+ */
+public class SslContextProvider
+{
+    private static final Logger log = Logger.get(SslContextProvider.class);
+    private static final String TLS_PROTOCOL = "TLS";
+
+    private final Optional<File> keystorePath;
+    private final Optional<String> keystorePassword;
+    private final Optional<File> truststorePath;
+    private final Optional<String> truststorePassword;
+    private final StandardErrorCode errorCode;
```

Comment:
Is this really needed? All the connectors using this class can simply throw the same error code and we are currently passing same error code from both places it is called from.